include/no_os_xspi.h

```diff
@@ -0,0 +1,255 @@
+/***************************************************************************//**
```

Comment:
I don't think we should add no_os_xpi.h/.c

We can just as well use no_os_spi.h module for the same purpose.
I've diffed your xspi.h to spi.h and it seems the only fundamental change is that you added a specifier for the number of lanes (single, bi, quad, octo). This could be easily done by adding a `lanes` member to the spi init_param and desc, which by default/uninitialized would mean single lane to be backwards compatible.

Then you'd proceed to implement your own implementation of the no_os_spi API in stm32_octospi for example, and populate the platform ops. Keep in mind that the same no_os_spi API could be implemented as xilinx_octalspi, so no_os_spi needs to be as generic as possible.

Reply:
Sure, this is a great idea and can be done. 

In STM HAL files, they have referred it to as XSPI driver and I was using the same convention as _xspi in this case. Will it be better to use the same or shall we proceed with _octospi?

Reply:
Maybe STM has very similar implementations for dual/quad/octo SPI's with very few register map differences and behavior, in which case it's better to make one stm32_xspi implementation that handles all of them, yes. But if they have wildly different implementations then it's better to keep them separate. I'm not very familiar with their offer.